bimmer_connected/utils.py

```diff
@@ -69,12 +69,7 @@ def deprecated(replacement: str = None):
 
     def decorator(func: "Callable[Concatenate[_T, _P], _R]") -> "Callable[Concatenate[_T, _P], _R | None]":
         def _func_wrapper(self: "_T", *args: "_P.args", **kwargs: "_P.kwargs") -> "_R | None":
```

Comment:
Not directly related to this PR, but you can improve the signature and typing of the `deprecated` decorator.

First, `self` isn't used aside from being passed as first argument to func. In this case, it's not necessary to split it out from `*args` and thus you also don't need to use `Concatenate`.

Second, `replacement` is typed as `str` but can also be `None`. Better to type it as `Optional[str]`.

Third, `deprecated` itself doesn't have a return type. Thus mypy will assume `Any` and accept everything.

```py
def deprecated(replacement: Optional[str] = None) -> Callable[
    ["Callable[_P, _R]"], "Callable[_P, _R | None]"
]:

    def decorator(func: "Callable[_P, _R]") -> "Callable[_P, _R | None]":
        def _func_wrapper(*args: "_P.args", **kwargs: "_P.kwargs") -> "_R | None":
            ...
            return func(*args, **kwargs)

        return _func_wrapper

    return decorator
```